### DISCUSSION TITLE
Should i make a pull requests for this table of content ?

### DISCUSSION THREAD

**Discussion by Th1nhNg0** [timlrx/tailwind-nextjs-starter-blog#229] [Ideas]
Just code this component, if u like it I can make a pull requests :D.
![image](https://media.giphy.com/media/Enot62pFkAjndo2hh4/source.gif?cid=790b761150728badb28effc218ea912f2da1125e1b6750c5&rid=source.gif&ct=g)

My blog: [https://th1nhng0.vercel.app/](https://th1nhng0.vercel.app/)

**Comment by vahidtakro** [comment]
> Just code this component, if u like it I can make a pull requests :D.
> ![image](https://camo.githubusercontent.com/d9e99042e7a335f8bfb74b9ffdb96610257b3bdf2a47527425094acc3c29a00a/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f456e6f74363270466b416a6e646f326868342f736f757263652e6769663f6369643d373930623736313135303732386261646232386566666332313865613931326632646131313235653162363735306335267269643d736f757263652e6769662663743d67)

Amazing!

**Reply by Th1nhNg0** [reply]
thanks 😄

**Reply by vahidtakro** [reply]
> thanks 😄

I'd like to have it, is your repo public? which files do I need to change?

**Reply by Th1nhNg0** [reply]
Yes my repo is public, you can check the component here:
https://github.com/Th1nhNg0/th1nhng0.vercel.app/blob/5e73a420828d82f01e7147512a2c3273c4ec19f8/layouts/PostLayout.js#L246-L296

**Reply by vahidtakro** [reply]
> Yes my repo is public, you can check the component here:
> https://github.com/Th1nhNg0/th1nhng0.vercel.app/blob/5e73a420828d82f01e7147512a2c3273c4ec19f8/layouts/PostLayout.js#L246-L296

perfect! I added, there is a problem, it only lists TOC with #, so if you don't have a #, there will not be a table of contents.

**Reply by Th1nhNg0** [reply]
Yeah, it's base on markdown heading so if you don't have any heading there will be no TOC.

**Reply by vahidtakro** [reply]
> Yeah, it's base on markdown heading so if you don't have any heading there will be no TOC.

Yes, but for example I have posts which have headings but with h3, h4, ... ##, ### ,
and when I add this on the top of MD files, it shows table of contents:
```
<TOCInline toc={props.toc} />
```
but there won't be anything on sticky table.
it requires contents with # headings, not ##, or ###

**Reply by Th1nhNg0** [reply]
Oh, I see. It's must be a bug. Fixed it btw :D 
https://github.com/Th1nhNg0/th1nhng0.vercel.app/blob/a2f11510dcc962dd03eef2a9e749ef38ca4ef975/layouts/PostLayout.js#L257-L262

**Reply by vahidtakro** [reply]
> Oh, I see. It's must be a bug. Fixed it btw :D
> https://github.com/Th1nhNg0/th1nhng0.vercel.app/blob/a2f11510dcc962dd03eef2a9e749ef38ca4ef975/layouts/PostLayout.js#L257-L262

well done! good job! it works very well now.
I think the style could be a bit better. like making it more smooth? 😀

**Reply by Th1nhNg0** [reply]
Thanks again!

> I think the style could be a bit better. like making it more smooth? 😀

I don't know xD. I'm not good at design, you can make a suggestion in my Github, I will make it better if I can. ヾ(≧▽≦*)o

**Comment by timlrx** [comment]
Nice! I probably would not be including this component at the moment, as there are just too many different possible layouts and components and it's starting to be a little cluttered. Would be happy to link your blog as an example though!

Going forward, if there's interest, we can think about restructuring the repo to make community contribution a little easier - maybe having a separate `community_pages` and `community_components` folder or even maintaining a monorepo. The ideal solution would provide a good base that does the markdown parsing part of things and allow you to opt in to various components / pages.

**Reply by Th1nhNg0** [reply]
Yeah, I had to edit a lot of code to customize it. So it would be great if we can do that ^-^
